pkg/interface/src/views/apps/chat/components/lib/chat-window.tsx

```diff
@@ -232,11 +232,14 @@ export class ChatWindow extends Component<ChatWindowProps, ChatWindowState> {
     const debouncedFetch = _.debounce(this.fetchMessages, 500).bind(this);
 
     envelopes
-      .concat(stationPendingMessages)
       .forEach((message) => {
         messages[message.number] = message;
       });
     
+    stationPendingMessages.sort((a, b) => a.when - b.when).forEach((message, index) => {
+      messages[mailboxSize + index] = message;
```

Comment:
I mean, indexes start at 0. So it'll start at the latest message.

Reply:
Damn, I'm sure that's it.